R/auditClassicalModelFairness.R

```diff
@@ -187,33 +221,48 @@ auditClassicalModelFairness <- function(jaspResults, dataset, options, ...) {
 
 .jfaFairnessSummaryTable <- function(dataset, options, fairnessContainer, jaspResults, ready, positionInContainer) {
   .jfaTableNumberUpdate(jaspResults)
-
+  metric <- .jfaFairnessGetMetricFromQuestion(options)
   if (!is.null(fairnessContainer[["summaryTable"]])) {
     return()
   }
-
-  metric <- .jfaFairnessGetMetricFromQuestion(options)
   title <- gettextf(
     "<b>Table %1$i.</b> Omnibus Test - %2$s",
     jaspResults[["tabNumber"]]$object,
     metric[["mainTitle"]]
   )
   tb <- createJaspTable(title = title)
   tb$position <- positionInContainer
-  tb$dependOn(options = c(.jfaFairnessCommonOptions(), "bayesFactorType"))
+  tb$dependOn(options = c(.jfaFairnessCommonOptions()))
   tb$addColumnInfo(name = "group", title = "", type = "string")
-  tb$addColumnInfo(name = "n", title = gettext("n"), type = "integer")
-  if (metric[["metric"]] != "dp") {
+  if (metric[["metric"]] != "pp") {
+    if (metric[["metric"]] != "eo") {
+      tb$addColumnInfo(name = "n", title = gettext("n"), type = "integer")
```

Comment:
I don't get why you add these columns separately for all three situations, they always need to be shows, right? Just add them once globally and then do all the if-elses. Same with the footnotes that can be shared across situations